### PR TITLE
player -> ply, guard on unlink

### PR DESCRIPTION
**Pull request by brandonsturgeon**

### CODE REVIEW

File: lua/entities/sent_spawnpoint/init.lua [hunk 1]
```diff
@@ -4,101 +4,103 @@ include( 'shared.lua' )
...
 end
 
-function unlinkPlayerFromSpawnPoint( player, spawnPoint )
-    player.LinkedSpawnPoint = nil
-    spawnPoint.LinkedPlayers[player] = nil
+function unlinkPlayerFromSpawnPoint( ply, spawnPoint )
+    if not IsValid(ply) then return end
+
+    ply.LinkedSpawnPoint = nil
+    spawnPoint.LinkedPlayers[ply] = nil
```
Comment (brandonsturgeon) [inline lua/entities/sent_spawnpoint/init.lua]: @PLally do you think this needs to run even if the player isn't valid?

Comment (PLally) [reply lua/entities/sent_spawnpoint/init.lua]: I don't think it will error if an invalid entity is passed,  maybe just check if its nil instead.

**Comment by brandonsturgeon** [comment]
In any case this needs a test

**Comment by brandonsturgeon** [comment]
@PLally @OverlordBleck Could someone give this guy a test? We should get this merged soon

**Pull request merged**